src/mono/wasm/runtime/hybrid-globalization/change-case.ts

```diff
@@ -13,15 +13,37 @@ export function mono_wasm_change_case_invariant(src: number, srcLength: number,
     const exceptionRoot = mono_wasm_new_external_root<MonoObject>(ex_address);
     try{
         const input = get_utf16_string(src, srcLength);
-        let result = toUpper ? input.toUpperCase() : input.toLowerCase();
+        const result = toUpper ? input.toUpperCase() : input.toLowerCase();
+
         // Unicode defines some codepoints which expand into multiple codepoints,
         // originally we do not support this expansion
-        if (result.length > dstLength)
-            result = input;
+        if (result.length <= dstLength)
```

Comment:
should this be compared with `input.length` instead of size of the buffer ? This will change behavior if there is buffer of different size.